src/pathogen_cluster/pathogen_cluster.py

```diff
@@ -0,0 +1,47 @@
+import hdbscan
+import pandas as pd
+import matplotlib.pyplot as plt
+import seaborn as sns
+import numpy as np
+
+def pathogen_cluster(args):
+
+    embedding_df = pd.read_csv(args.embedding, index_col=0)
+
+    clustering_parameters = {
+        **({"min_cluster_size": args.min_size} if args.min_size is not None else {}),
+        **({"min_samples": args.min_samples} if args.min_samples is not None else {}),
+        **({"cluster_selection_epsilon": args.distance_threshold} if args.distance_threshold is not None else {})
+    }
+
+    clusterer = hdbscan.HDBSCAN(**clustering_parameters)
+    clusterer_default = hdbscan.HDBSCAN()
+
+    clusterer.fit(embedding_df)
+    clusterer_default.fit(embedding_df)
+    embedding_df["label"] = clusterer.labels_.astype(str)

```

Comment:
We'll want a way to set different names for these cluster labels, so we can run multiple cluster commands on the same embedding file and merge the final results without conflicts. We might want to add a command line parameter like `--label-attribute` to support this functionality.